lib/engine/game/base.rb

```diff
@@ -477,11 +477,21 @@ def init_company_abilities
 
           case (share = ability[:share].to_s)
           when 'random-president'
-            share = @corporations[rand % @corporations.size].shares[0]
+            corporation = @corporations[rand % @corporations.size]
+            share = corporation.shares[0]
             ability[:share] = share
-            corporation = share.corporation
             company.desc = "#{company.desc} The random corporation in this game is #{corporation.name}."
             @log << "#{company.name} comes with the president's share of #{corporation.name}"
+          when 'random'
+            exclusions = ability[:'random-share-exclude']&.split
+            corporation = nil
+            while corporation.nil? || exclusions&.include?(corporation.name)
+              corporation = @corporations[rand % @corporations.size]
+            end
+            share = corporation.shares[1] # not president
```

Comment:
it's probbaly safe but if you want to be more robust, corporation.shares.find { |s| !s.president }